src/main/java/org/gridsuite/explore/server/services/ExploreService.java

```diff
@@ -277,4 +277,28 @@ public void assertCanCreateCase(String userId) {
             }
         }
     }
+
+    public void updateElement(UUID id, ElementAttributes elementAttributes, String userId) {
+        directoryService.updateElement(id, elementAttributes, userId);
+        ElementAttributes elementsInfos = directoryService.getElementInfos(id);
+        // send notification if the study name was updated
+        if (STUDY.equals(elementsInfos.getType()) && StringUtils.isNotBlank(elementAttributes.getElementName())) {
+            studyService.notifyStudyUpdate(id, userId);
+        }
+    }
+
+    public void moveElementsDirectory(List<UUID> elementsUuids, UUID targetDirectoryUuid, String userId) {
+        directoryService.moveElementsDirectory(elementsUuids, targetDirectoryUuid, userId);
+        //send notification to all studies
+        List<ElementAttributes> elementsAttributes = directoryService.getElementsInfos(elementsUuids, null);
```

Comment:
That's pity to make this request to the directory-server, just to get the type of the element...

Reply:
> That's pity to make this request to the directory-server, just to get the type of the element...

I don't see an other way to do it since the information is stored in the directory server.

Reply:
Will fix later when removing `/notification` endpoint from study-server